scripts/DB.js

```diff
@@ -1,48 +1,109 @@
 /**
  * Connections to the Database are made here
  */
-import patients from "../stubs/patients";
-import tasks from "../stubs/tasks";
-// Note: Look into pools? https://www.npmjs.com/package/mysql#pooling-connections
+require("dotenv").config();
+var mysql = require("mysql");
+const DBConnection = require("./DBConnection");
 
-const DB = {
-  /** ----- Tasks ----- */
-  retrieveTask: function (taskID) {
-    const task = tasks[taskID];
-    if (patient === undefined)
-      throw Error(`Task '${taskID}' does not exist in the Database`);
+var connection = mysql.createConnection({
+  host: process.env.DB_HOST,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASSWORD,
+  database: process.env.DB,
+});
 
-    return task;
-  },
+// connection.connect(function (err) {
+//   if (err) throw err;
+//   console.log("Connected!");
+// });
 
-  retrieveTask: function (taskID) {
-    const task = tasks.find((task) => task.id === taskID);
-    if (task === undefined)
-      throw Error(`Task '${taskID}' does not exist in the Database`);
+const DB = {
+  /** ----- Tasks ----- */
+  retrieveTask: function (taskID, callback) {
+    // need to error handle now
+    const sql = `SELECT * FROM Tasks WHERE taskID = ${taskID}`;
 
-    return task;
+    DBConnection.query(sql, (err, result) => {
+      if (err) throw err;
+      if (result.length > 0){
+        console.log(`DB.retrieveTask(${sql}):`, result);
+        callback(result);
+      }
+      else {
+        callback('TaskID does not exist in DB');
```

Comment:
I think it would be best to `throw` an error here `throw Error("message")`. That way we could use a `try/catch` in our routes. Using `callback(error)` will make it more complicated to read errors.